src/lua.cpp

```diff
@@ -242,6 +245,80 @@ static int disassembleTeakThunk(lua_State* L) {
 	return 1;
 }
 
+#ifndef __ANDROID__
+// Haptics functions
+
+namespace Haptics {
+	std::unique_ptr<ExternalHapticsManager> hapticsManager = nullptr;
+
+	static int initHapticsThunk(lua_State* L) {
+		if (hapticsManager == nullptr) {
+			hapticsManager.reset(new ExternalHapticsManager());
+		}
+
+		return 0;
+	}
+
+#define HAPTICS_THUNK(func)                \
+	static int func##Thunk(lua_State* L) { \
+		if (hapticsManager != nullptr) {   \
+			hapticsManager->func();        \
+		}                                  \
+                                           \
+		return 0;                          \
+	}
+
+	HAPTICS_THUNK(startScan)
+	HAPTICS_THUNK(stopScan)
+	HAPTICS_THUNK(connect)
+	HAPTICS_THUNK(requestDeviceList)
+	HAPTICS_THUNK(getDevices)
+	HAPTICS_THUNK(getSensors)
+	HAPTICS_THUNK(stopAllDevices)
+	#undef HAPTICS_THUNK
+
+	static int sendScalarThunk(lua_State* L) {
+		const int device = (int)lua_tonumber(L, 1);
+		const auto value = lua_tonumber(L, 2);  
+
+		if (hapticsManager != nullptr) {
+			hapticsManager->sendScalar(device, value);
+		}
+
+		return 2;
```

Comment:
Magic number

Reply:
Sir this is how Lua thunks work, you return the number of parameters expected